concordium-consensus/src/Concordium/GlobalState/Persistent/Instances.hs

```diff
@@ -105,6 +116,26 @@ data PersistentInstanceV (v :: Wasm.WasmVersion) = PersistentInstanceV {
     pinstanceHash :: H.Hash
 }
 
+migratePersistentInstanceV ::
+    forall v t m.
+    ( 
+      BlobStorable m (ModuleV v)
+    , BlobStorable (t m) (ModuleV v)
+    , MonadTrans t
+    ) =>
+    PersistentInstanceV v ->
+    t m (PersistentInstanceV v)
+migratePersistentInstanceV PersistentInstanceV{..} = do
+  newInstanceParameters <- migrateReference return pinstanceParameters
+  newInstanceModuleInterface <- migrateReference return pinstanceModuleInterface
```

Comment:
This will be problematic, since it will make a new copy of the module (for each instance). Instead, we should look up the module in the modules table (assuming that it is already migrated) and use that reference to ensure sharing.

Reply:
Fixed. No module migration is done here now. We do it as for "cache" now.